integration/query_test.go

```diff
@@ -736,6 +745,96 @@ func TestQueryCommandBatchSize(t *testing.T) {
 	}
 }
 
+func TestQueryCommandSingleBatch(t *testing.T) {
+	t.Parallel()
+	ctx, collection := setup.Setup(t)
+
+	docs := generateDocuments(0, 5)
+	_, err := collection.InsertMany(ctx, docs)
+	require.NoError(t, err)
+
+	for name, tc := range map[string]struct { //nolint:vet // used for testing only
+		batchSize    any  // optional, nil to leave batchSize unset
+		singleBatch  any  // optional, nil to leave singleBatch unset
+		cursorClosed bool // optional, set true for expecting cursor to be closed
```

Comment:
I think that should be in a second group – to separate setup fields from expectations.

We probably should do the same in other related tests